addons/project/migrations/9.0.1.1/openupgrade_analysis_work.txt

```diff
@@ -0,0 +1,162 @@
+# Concerns module Mail (has to be done in another migration script specifically for that module) ---> Nothing to do
+
+project      / project.project          / state (selection)             : selection_keys is now '['cancelled', 'close', 'draft', 'open', 'pending']' ('['cancelled', 'close', 'draft', 'open', 'pending', 'template']')
+# Done ---> 'template' is converted to 'draft'
+# TODO - does this behavior make sense on the functional level?
+
+project      / project.project          / user_id (many2one)            : NEW relation: res.users
+# TODO - does someone know where user_id comes from?
+
+project      / project.project          / website_message_ids (one2many): NEW relation: mail.message
+# Field doesn't exist neither in v8 or in v9...weird! ---> Nothing to do
+
+project      / project.tags             / color (integer)               : NEW
+project      / project.tags             / name (char)                   : NEW required: required
+# New fields ---> Nothing to do
+
+project      / project.task             / active (boolean)              : not a function anymore
+# Field content is still a boolean ---> Nothing to do
+
+project      / project.task             / attachment_ids (one2many)     : NEW relation: ir.attachment
+# Nothing to do
+
+project      / project.task             / categ_ids (many2many)         : relation is now 'project.tags' ('project.category')
+# Renamed project_category to project_tags (note: column color is missing from project_category but should be created automatically)
+
+project      / project.task             / categ_ids (many2many)         : was renamed to tag_ids [nothing to to]
+# Field was renamed (has oldname) ---> Nothing to do
+
+project      / project.task             / date_assign (datetime)        : NEW
+# New field ---> Nothing to do
+
+project      / project.task             / description (text)            : type is now 'html' ('text')
+# Converted 'description' value to HTML
+
+project      / project.task             / displayed_image_id (many2one) : NEW relation: ir.attachment
+# New relation AND new field ---> Nothing to do
+
+project      / project.task             / kanban_state (selection)      : now required, default = normal
+# Field is now required and default is the same as before ---> Nothing to do
+
+project      / project.task             / message_follower_ids (many2many): not a function anymore
+project      / project.task             / message_follower_ids (many2many): relation is now 'mail.followers' ('res.partner')
+project      / project.task             / message_follower_ids (many2many): type is now 'one2many' ('many2many')
+# Concerns module Mail (has to be done in another migration script specifically for that module) ---> Nothing to do
+
+project      / project.task             / priority (selection)          : selection_keys is now '['0', '1']' ('['0', '1', '2']')
+# Done ---> '2' is converted to '1' and '1' is converted to '0'
```

Comment:
We actually have a module that extends the stars to show 5 instead of 3. We might publish it for those that want to keep the 3 star priority system. We need to preserve the values in this field though. Should we build a module that modifies the upgrade script or should we store the values in another field in order to restore them after the upgrade?